### ISSUE TITLE
Upgrade Windshaft

### ISSUE THREAD

**Issue by rajadain** [WikiWatershed/model-my-watershed#3091]
We are using an old version of Windshaft:

https://github.com/WikiWatershed/model-my-watershed/blob/ab0389e6cb5bc27c2f8c464c537a898ba17362c3/src/tiler/package.json#L9

The current version is [4.13.1](https://github.com/CartoDB/Windshaft/releases/tag/4.13.1).

Our version of Windshaft has [a bug](https://github.com/CartoDB/Windshaft/issues/310) which causes improper internal caching of UtfGrid tile requests, which can cause a first request of a uniform tile (where all the pixels are the same id) to be cached, and when a second request for a uniform tile in a different region is accessed, which should have all the pixels of a different id, the first is returned.

This causes bugs like #3082, which can severely impede app's usability.

For a guide on how to manage upgrading Windshaft, which may involve extensive changes to the API, see here: https://github.com/OpenTreeMap/otm-tiler/pull/116

**Comment by hectcastro** [comment]
I believe that the version of `libgif` that Windshaft (or one of its dependencies) expects is no longer available on Ubuntu 16.04. When we upgrade, we will need to upgrade to a version that can use `libgif` v5:

```
Jun 12 18:26:05 tiler envdir[31790]: module.js:356
Jun 12 18:26:05 tiler envdir[31790]:   Module._extensions[extension](this, filename);
Jun 12 18:26:05 tiler envdir[31790]:                                ^
Jun 12 18:26:05 tiler envdir[31790]: Error: libgif.so.4: cannot open shared object file: No such file or directory
Jun 12 18:26:05 tiler envdir[31790]:     at Module.load (module.js:356:32)
Jun 12 18:26:05 tiler envdir[31790]:     at Function.Module._load (module.js:312:12)
Jun 12 18:26:05 tiler envdir[31790]:     at Module.require (module.js:364:17)
Jun 12 18:26:05 tiler envdir[31790]:     at require (module.js:380:17)
Jun 12 18:26:05 tiler envdir[31790]:     at Object.<anonymous> (/opt/tiler/node_modules/windshaft/node_modules/canvas/lib/bindings.js:2:18)
Jun 12 18:26:05 tiler envdir[31790]:     at Module._compile (module.js:456:26)
Jun 12 18:26:05 tiler envdir[31790]:     at Object.Module._extensions..js (module.js:474:10)
Jun 12 18:26:05 tiler envdir[31790]:     at Module.load (module.js:356:32)
Jun 12 18:26:05 tiler envdir[31790]:     at Function.Module._load (module.js:312:12)
Jun 12 18:26:05 tiler envdir[31790]:     at Module.require (module.js:364:17)
```

**Comment by rajadain** [comment]
May have to use different versions of `node` for app and tiler, as done here: https://github.com/OpenTreeMap/otm-cloud/commit/5a458b61a3c004d5a27fab3a3f0f64520aefbf60. Keep `app` where it is, and upgrade `tiler` to latest LTS.
